pkg/sinks/s3/s3.go

```diff
@@ -75,8 +80,38 @@ func (s *S3Sink) Init(ctx context.Context, format formats.Format, compression kt
 	if s.Bucket == "" {
 		return fmt.Errorf("Not writing to s3 -- no bucket set, use -s3_bucket flag")
 	}
-	sess := session.Must(session.NewSession())
-	s.client = s3manager.NewUploader(sess)
+
+	if s.config.assumeRoleARN != "" {
+		// Getting credentials from assume role ARN
+		// sess_tmp := session.Must(session.NewSession())
+		sess_tmp := session.Must(
+			session.NewSessionWithOptions(session.Options{
+			  SharedConfigState: session.SharedConfigEnable,
+			}),
+		)
+		creds := stscreds.NewCredentials(sess_tmp, s.config.assumeRoleARN)
+		_ , err := creds.Get()
+		if err != nil {
+			return fmt.Errorf("Assume Role ARN doesn't work. ARN: %q", s.config.assumeRoleARN)
+		}
+		// Creating a new session from assume role
+		sess, err := session.NewSession(
+			&aws.Config{
+				Region: aws.String(s.config.Region), 
+				Credentials: creds,
+			},
+		)
+		if err != nil {
+			return fmt.Errorf("Session is not created with region %q", s.config.Region)
+		}
+		// sess := session.Must(session.NewSession(&aws.config{Region: aws.String(s3Region), Credentials: creds}))
+		s.client = s3manager.NewUploader(sess)
+	} else {
+		sess := session.Must(session.NewSession())
+		s.client = s3manager.NewUploader(sess)
+	}
+
+	// s.client = s3manager.NewUploader(sess)
```

Comment:
Take this comment out, along with the other commented line.

Reply:
Sure, working on it